docs/backend-system/building-plugins-and-modules/01-index.md

```diff
@@ -168,7 +168,61 @@ Whenever you want to allow modules to configure your plugin dynamically, for
 example in the way that the catalog backend lets catalog modules inject
 additional entity providers, you can use the extension points mechanism. This is
 described in detail with code examples in [the extension points architecture
-article](../architecture/05-extension-points.md).
+article](../architecture/05-extension-points.md), while the following is a more
+slim example of how to implement an extension point for a plugin:
+
+```ts
+import { createExtensionPoint } from '@backstage/backend-plugin-api';
+
+// This is the extension point interface, which is how modules interact with your plugin.
+export interface ExamplesExtensionPoint {
+  addExample(example: Example): void;
+}
+
+// This is the extension point reference that encapsulates the above interface.
+export const examplesExtensionPoint =
+  createExtensionPoint<ExamplesExtensionPoint>({
+    id: 'example.examples',
+  });
+
+// This is the implementation of the extension point, which is internal to your plugin.
+class ExamplesExtension implements ExamplesExtensionPoint {
+  #examples: Example[] = [];
+
+  addActions(example: Examples): void {
```

Comment:
```suggestion
  addExample(example: Examples): void {
```

Reply:
perhaps `example: Example` instead of `example: Examples`